SIGNALESP/SIGNALESP.ino

```diff
@@ -247,17 +247,39 @@ void loop() {
 		if (!command_available) { cmdstring = ""; }
 		blinkLED = true;
 	}
-	yield();
+
+//	yield();  // crep
 	if (fifousage < FiFo.count())
 	  fifousage = FiFo.count();
-	while (FiFo.count()>0) { //Puffer auslesen und an Dekoder uebergeben
-
-		aktVal = FiFo.dequeue();
-		state = musterDec.decode(&aktVal);
-		if (state) blinkLED = true; //LED blinken, wenn Meldung dekodiert
-		yield();
-	}
-
+  	while (FiFo.count()>0) { //Puffer auslesen und an Dekoder uebergeben
+  		aktVal = FiFo.dequeue();
+	  	state = musterDec.decode(&aktVal);
+		  if (state) blinkLED = true; //LED blinken, wenn Meldung dekodiert
+//		  yield();  // crep
```

Comment:
finde ich nicht so gut. Wir brauchen das yield() auch für die Interrupts.

Reply:
Es sind sehr viele Aufrufe von yield() im Code. wenn nun eine Nachricht geschrieben wurde, dann wird durch yield() eh in den Wifi-Stack gewechselt und die Timings sind dann sowieso murks.
ich würde eher versuchen, an dedizierten Stellen (z.B. MSG_PRINTLN-Macro) so wenig wie möglich yield()-Aufrufe zu benutzen. 

Bei mir funktioniert ITv1 senden und empfangen und SOMFY-Empfang.

Reply:
Ich habe yield() absichtlich so oft eingefügt, damit auch sehr oft Hintergrundoperationen ausgeführt werden. Der esp8266 hat keinen HW Interrupt, er hat nur einen SW Interrupt.
Ob yield noch mal in der main funktion aufgerufen wird, weiss ich nicht. Aber yield() arbeitet auch immer nur ein einzelnes Kommando ab. 

Der Bezug zu Timings ist mir nun aktuell noch nicht transparent.

Reply:
Bei jedem Aufruf von yield() kann eine unbestimmte Menge an Zeit vergehen. Wenn z.B. auf der Netzwerkschiene gesendet oder empfangen wurde sind schätzungsweise 50ms weg. Interrupts werden wahrscheinlich deutlich billiger.

Reply:
Was meinst Du mit Interrupts werden billiger?

Reply:
Zeit, die vergeht, während der Interrupt abgearbeitet wird.

Reply:
Ich kann dir leider nicht folgen. yield() führt eine  hintergrund Operationen aus. Wie lange die dauert ist ungewiss.
Es muss sichergestellt sein, dass yield() auch dann aufgerufen wird, wenn der ESP nichts empfängt. Sonst startet der Watchdog den uC neu und der ESP kann auch nicht auf eingehende Befehle reagieren.

Reply:
Habe mir den ESP8266 core quellcode angesehen:
https://github.com/esp8266/Arduino/blob/master/cores/esp8266/core_esp8266_main.cpp

vom loop_wrapper wird 
    loop();
    run_scheduled_functions();
    esp_schedule();
aufgerufen. Damit dürfte erst mal klar sein, dass yield in der loop nicht notwendig ist. Nur wer den interrupt aufruft habe ich noch nicht raus

Reply:
schau mal in [core_esp8266_wiring_digital.c](https://github.com/esp8266/Arduino/blob/ba0e049b83a7d012c857c3c25c0b148a76fd17a1/cores/esp8266/core_esp8266_wiring_digital.c#L120)

Reply:
Das habe ich gefunden, aber was den handler aufruft weiss ich nicht.
Der ESP8266 hat keinen hardware Interrupt. Also muss es ein scheduler oder so etwas machen.

Reply:
[core_esp8266_wiring_digital.c](https://github.com/esp8266/Arduino/blob/ba0e049b83a7d012c857c3c25c0b148a76fd17a1/cores/esp8266/core_esp8266_wiring_digital.c#L197)